src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8471,6 +8472,37 @@ private Symbol GetSymbolOrMethodOrPropertyGroup(LookupResult result, SyntaxNode
             return ResultSymbol(result, plainName, arity, node, diagnostics, false, out wasError, qualifierOpt);
         }
 
+        /// <summary>
+        /// Interprets a LookupResult as either a singular symbol or a method/property group.
+        /// It is "strict" in the sense that if the members in the result have different kinds,
+        /// an error result is always returned.
+        /// </summary>
+        private Symbol GetSymbolOrMethodOrPropertyGroupStrict(LookupResult result, SyntaxNode node, string plainName, int arity, ArrayBuilder<Symbol> methodOrPropertyGroup, BindingDiagnosticBag diagnostics, NamespaceOrTypeSymbol qualifierOpt)
+        {
+            Debug.Assert(!methodOrPropertyGroup.Any());
+
+            node = GetNameSyntax(node) ?? node;
+
+            Debug.Assert(result.Kind != LookupResultKind.Empty);
+            Debug.Assert(!result.Symbols.Any(s => s.IsIndexer()));
+
+            SymbolKind kind = result.Symbols[0].Kind;
+            if (!result.Symbols.All(s => s.Kind == kind))
```

Comment:
Consider using `static` lambda (via another overload of `All`).

Reply:
We can't, as there's a capture of `kind`. I'll rewrite as `foreach` instead. Thanks

Reply:
I thought there's another overload of `All` that takes additional arg, like

```suggestion
            if (!result.Symbols.All(static (s, kind) => s.Kind == kind, kind))
```

but maybe it doesn't exist